packages/ra-ui-materialui/src/field/BooleanField.stories.tsx

```diff
@@ -19,6 +19,31 @@ export const Basic = (props: BooleanFieldProps) => {
     );
 };
 
+type Post = {
+    published: boolean;
+    reported: boolean;
+    title: string;
+};
+
+export const Typed = (props: Omit<BooleanFieldProps, 'source' | 'sortBy'>) => {
+    const [published, setPublished] = React.useState(true);
+    return (
+        <Stack direction="row">
+            <input
+                type="checkbox"
+                checked={published}
+                onChange={e => setPublished(e.target.checked)}
+            />
+            <BooleanField<Post>
+                record={{ published }}
```

Comment:
I see a TS error at this line: `Type '{ published: boolean; }' is missing the following properties from type 'Post': reported, title`

Unfortunately, when I fix it, I get another error which I have not been able to fix: `Type 'unknown' is not assignable to type 'Post | undefined'`